src/archutils/Win32/GraphicsWindow.cpp

```diff
@@ -425,41 +425,38 @@ void GraphicsWindow::CreateGraphicsWindow( const VideoModeParams &p, bool bForce
 /** @brief Shut down the window, but don't reset the video mode. */
 void GraphicsWindow::DestroyGraphicsWindow()
 {
-	if( g_HDC != nullptr )
+	// If we were in fullscreen, release the display mode before destroying the window
+	if( g_hWndMain && !g_CurrentParams.windowed )
+	{
+		ChangeDisplaySettingsEx(g_CurrentParams.sDisplayId, nullptr, nullptr, 0, nullptr);
+	}
+
+	// Release and destroy the window and its resources.
+	if( g_HDC )
```

Comment:
I'd opt to keep the != nullptr since it conveys it's a pointer (which handles are)